### PR TITLE
Deprecate PHP 5, add PHP 7 features

### PR DESCRIPTION
**Pull request by gregurco**
- [x] Bug fix
- [ ] New feature
- [ ] BC breaks
- [ ] Deprecations
- [x] Tests pass

Fixed tickets: [#96]
License: MIT

What I did:
- [x] Add parameters type hinting
- [x] array() to []
- [x] Add classes to use
- [x] Return type to methods
- [x] Readme
- [x] Fix tests
- [x] Fix comments
- [x] Fix typos
- [x] Fix code style

### CODE REVIEW

**Comment by gregurco** [comment]
@florianpreusner could you check this PR? I think it's done. I also added symfony 3.3.* in travis. All checks passes.

File: Tests/DataCollector/HttpDataCollectorTest.php [hunk 1]
```diff
@@ -173,11 +190,11 @@ public function testCallCount()
      * @version 2.1
      * @since   2015-06
      *
-     * @covers  EightPoints\Bundle\GuzzleBundle\DataCollector\HttpDataCollector::getErrorCount
+     * @covers \EightPoints\Bundle\GuzzleBundle\DataCollector\HttpDataCollector::getErrorCount
      */
     public function testErrorCount()
     {
-        // implement me
+        $this->markTestSkipped('must be implemented.');
```
Comment (florianpreusner) [inline Tests/DataCollector/HttpDataCollectorTest.php]: 👍

File: Tests/DataCollector/HttpDataCollectorTest.php [hunk 2]
```diff
@@ -186,10 +203,10 @@ public function testErrorCount()
      * @version 2.1
      * @since   2015-06
      *
-     * @covers  EightPoints\Bundle\GuzzleBundle\DataCollector\HttpDataCollector::getTotalTime
+     * @covers \EightPoints\Bundle\GuzzleBundle\DataCollector\HttpDataCollector::getTotalTime
      */
     public function testTotalTime()
     {
-        // implement me
+        $this->markTestSkipped('must be implemented.');
```
Comment (florianpreusner) [inline Tests/DataCollector/HttpDataCollectorTest.php]: 👍

File: Tests/Events/PostTransactionEventTest.php [hunk 3]
```diff
@@ -33,16 +35,16 @@ public function testConstruct()
...
+     * @covers \EightPoints\Bundle\GuzzleBundle\Events\PostTransactionEvent::setTransaction
+     * @covers \EightPoints\Bundle\GuzzleBundle\Events\PostTransactionEvent::getTransaction
      */
-    public function testTranscation()
+    public function testTransaction()
     {
         $statusCode = 204;
-        $response   = $this->createMock('GuzzleHttp\Psr7\Response');
-        $postEvent  = new PostTransactionEvent($response, null);
+        $response   = $this->createMock(Response::class);
```
Comment (florianpreusner) [inline Tests/Events/PostTransactionEventTest.php]: 👍

**Pull request merged**